pkg/cmd/pulumi/templates/org.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2016-2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package templates
+
+import (
+	"archive/tar"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend"
+	"github.com/pulumi/pulumi/pkg/v3/backend/display"
+	cmdBackend "github.com/pulumi/pulumi/pkg/v3/cmd/pulumi/backend"
+	pkgWorkspace "github.com/pulumi/pulumi/pkg/v3/workspace"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/apitype"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/logging"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+func (s *Source) getOrgTemplates(
+	ctx context.Context, templateName string,
+	interactive bool, wg *sync.WaitGroup,
+) {
+	ws := pkgWorkspace.Instance
+	project, _, err := ws.ReadProject()
+	if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
+		s.addError(err)
+		return
+	}
+
+	b, err := cmdBackend.CurrentBackend(ctx, ws, cmdBackend.DefaultLoginManager, project, display.Options{
+		Color:         cmdutil.GetGlobalColorization(),
+		IsInteractive: interactive,
+	})
+	if err != nil {
+		if !errors.Is(err, backend.MissingEnvVarForNonInteractiveError{}) {
+			s.addError(err)
+		}
+		return
+	}
+	if !b.SupportsTemplates() {
+		return
+	}
+
+	logging.Infof("Listing Org Templates from the cloud")
+	user, orgs, _, err := b.CurrentUser()
+	if err != nil {
+		s.addError(err)
+		return
+	} else if user == "" {
+		return // No current user - so don't proceed.
+	}
+
+	alreadySeenSourceURLs := map[string]struct{}{}
+
+	handleOrg := func(org string) {
+		defer wg.Done()
+		logging.Infof("Checking for templates from %q", org)
+		orgTemplates, err := b.ListTemplates(ctx, org)
+		if apiError := new(apitype.ErrorResponse); errors.As(err, &apiError) {
+			// This is what happens when we try to access org templates for an org that hasn't enabled org templates.
+			if apiError.Code == 402 {
+				logging.Infof("%q does not have access to org templates (code=%d)", org, apiError.Code)
```

Comment:
If they don't have access to org templates shouldn't we just silently not use them?  This doesn't strike me as a great user experience for those not having access to them.  This is especially true if the user is part of multiple orgs and only one of them supports templates.